tasks.py

```diff
@@ -938,13 +938,23 @@ def pending_travis_build():
         if b["tag"]:
             tag = b["tag"]["name"]
 
-        return {
-            "branch": b["branch"]["name"],
-            "commit_id": b["commit"]["sha"],
-            "tag": tag,
-            "id": b["id"],
-        }
-
+        if branch is None:
+            return {
+                "branch": b["branch"]["name"],
+                "commit_id": b["commit"]["sha"],
+                "tag": tag,
+                "id": b["id"],
+                "latest_build_commit_id": b['branch']['last_build']['commit']['sha'] if "last_build" in b['branch'] and 'commit' in b['branch']['last_build'] else b["commit"]["sha"]
+            }
+        elif branch == b["branch"]["name"]:
+            return {
+                "branch": b["branch"]["name"],
+                "commit_id": b["commit"]["sha"],
+                "tag": tag,
+                "id": b["id"],
+                "latest_build_commit_id": b['branch']['last_build']['commit']['sha'] if "last_build" in b[
+                    'branch'] and 'commit' in b['branch']['last_build'] else b["commit"]["sha"]
+            }
```

Comment:
Can we also add an 
```python
else:
    raise NotImplementedError
```

Reply:
They return the same but because it's a for loop, the first condition returns the first element always and the second  returns the build for that particular branch. We don't want to raise an exception but rather return None when there are no relevant builds (the behavior now) but made it explicit.

Reply:
Not sure I fully understand, it seems that the dictionary defined in lines 941-947 and on lines 950-957 are identical. I.e. I could change the first statement to `if branch is None or branch == b["branch"]["name"]`, remove the `elif`, and have the same result.

Reply:
Done